openmsupply/application-manager-web-app
# #385 update use get review structure to get review responses for lvl 1 review (first review)

Fixes: #385


### Changes

* added reviewResponseFragment
* getAllResponses also gets a fragment of review response (to show for applicant during LOQ, still needs visibility filter), this was done because of changes to getEvaluatedResponses
* new query of getReviewNew (needs reviewAssignment and userid), review response that we get for applicationResponse are filtered such that only submitted or draft ones belonging to existing user are returned. Also we do orgerBy TIME_CREATED for reviewResponses
* add helper to add elementsById to structure (see it as useful field, it is held by reference so when it changed, corresponding elements will change in structure)
* added getGroupedReviewResponses, returns key templateElementId, and values = reviewResponses
* added addOwnedReviewResponse helper, will add latest owned review response to structure
* changed addEvaluatedResponseToStructure to all add review responses to response
* useGetFullReviewStructure hook adds owned review response to the structure

### Testing

* use 385-testing branch
* use this query in PG to see what review_assignments are there (for what user and what status)
```
select application.id as appid, application.serial, review_assignment.id as raid, review.reviewer_id as reviewer_id, review_response.time_created, review_response.status from application join review_assignment on review_assignment.application_id = application.id join review on review.application_id = application.id join review_response on review_response.review_id = review.id
```

See output of new structure in `applicationNew/${serial}/reviewTest` and adjust user (switch user) and reviewAssignment in `ReviewPageWrapperTestNew` to see different results
